test/udp_sender.js

```diff
@@ -249,4 +250,20 @@ describe('udp sender should', () => {
         assert.equal(response.err, false);
         assert.equal(response.numSpans, 0);
     });
+
+    it ('flush gracefully handles errors emitted by socket.send', () => {
+        sender._client.on('error', err => {
+            assert.equal(err.message, 'getaddrinfo ENOTFOUND foo.bar.com')
```

Comment:
Nit: this isn't really testing the new error handler you're adding. The test should simply try to send a span when the sender has the wrong agent address, and without the new handler it should throw uncaught error, but with the handler just log it to console and succeed.

Also, I am not sure you would actually get uncaught error here since the default reporter buffers the spans.

The ideal test would be to validate that `console.log` was called with expected error string.